### ISSUE TITLE
Add optional path extention

### ISSUE THREAD

**Issue by PaulMaly** [matthiaaas/express-file-routing#1]
Hi! Thanks for this great package.

Can you add the ability to use URLs ended with file extension? For example:

```
/posts.json => /routes/posts.js
/posts/1.json =>  /routes/posts/:id.js
```
etc.

It's very useful if we use SPA/SSR application as a part of the same express app as this middleware. So, this application also has some routing, and these two routings may start to conflict.

**Comment by matthiaaas** [comment]
hey Paul, if I got you right, you can do this already by naming your route files `/routes/posts.json.js` for instance. Is that helpful?

**Comment by PaulMaly** [comment]
Actually, I already checked this variant and unfortunately, it's not working properly. Let's imagine we've 3 routes:

```
 / => /routes/index.js
/posts => /routes/posts/index.js
/posts/1 => /routes/posts/:id.js
```

If just to rename these files with `.json` in filename we'll get following result:

```
 / => /routes/index.json.js (not working, because index-file won't autopmatically resolved)
/posts => /routes/posts/index.json.js  (not working, because index-file won't autopmatically resolved)
/posts/1 => /routes/posts/:id.json.js  (works well)
```

I think it's because the index file won't be automatically resolved.

**Comment by matthiaaas** [comment]
Yes, that's right. Currently, you aren't able to use `index` files for routes with an additional extension. But is it a good idea to resolve any `index.*` route file name?

**Comment by matthiaaas** [comment]
@PaulMaly fixed in 14d34f9

**Comment by matthiaaas** [comment]
Available since [v1.1.1](https://github.com/matthiaaas/express-file-routing/releases/tag/v1.1.1) on npm registry as of now
